### PR TITLE
Adds a special poster type that changes with holiday

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a poster type that changes with the current holiday(s). The name, desc, and icon_state are defined in the holiday datums, in order to keep it organized :)

## Why It's Good For The Game
FESTIVE

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
(done by manually modifying SSevents)

</details>

## Changelog
:cl:
add: festive posters
/:cl:

### CODE REVIEW

File: code/game/objects/effects/contraband.dm [hunk 1]
```diff
@@ -662,4 +665,21 @@
 	desc = "A Syndicate-commissioned poster that uses Syndie Moth(TM?) to tell the viewer to keep the nuclear authentication disk unsecured. No, we aren't doing that. It's signed by 'AspEv'."
 	icon_state = "poster49"
 
+/obj/structure/sign/poster/official/festive
+	name = "Celebrate the Day!"
+	desc = "A poster informing the reader every day is a gift, and not to waste it."
+	icon = 'icons/obj/festive_posters.dmi'
+	icon_state = "generic"
+
+/obj/structure/sign/poster/official/festive/Initialize()
```
Comment (Zonespace27) [inline code/game/objects/effects/contraband.dm:674]: Call parent

File: code/game/objects/effects/contraband.dm [hunk 2]
```diff
@@ -662,4 +665,21 @@
...
+	if(!length(SSevents.holidays))
+		return
+	var/cele_holiday = pick(SSevents.holidays)
```
Comment (Zonespace27) [inline code/game/objects/effects/contraband.dm:678]: You don't need to make this var

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/contraband.dm (hunk 2 above)
Comment (etherware-novice) [reply code/game/objects/effects/contraband.dm:678]: i feel like the line would be way too long if i left the pick function in line

File: code/game/objects/effects/contraband.dm [hunk 3]
```diff
@@ -78,6 +78,9 @@
 
 	var/obj/structure/sign/poster/selected = pick(approved_types)
 
+	if(SSevents.holidays && prob(30))  // more festive posters on festive days!
+		selected = /obj/structure/sign/poster/official/festive
```
Comment (itsmeow) [inline code/game/objects/effects/contraband.dm:82]: this should only apply if the holiday has a poster

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/contraband.dm (hunk 3 above)
Comment (etherware-novice) [reply code/game/objects/effects/contraband.dm:82]: it has a generic poster varient so it doesnt mess with the random poster code too much ^^

**Pull request closed without merging**